x-pack/solutions/security/plugins/security_solution/public/data_view_manager/hooks/use_sync_url_state.test.ts

```diff
@@ -0,0 +1,118 @@
+  });
+
+  it('should not dispatch or update url param if newDataViewPickerEnabled is true', () => {
+    mockUseIsExperimentalFeatureEnabled.mockReturnValue(true);
+    renderHook(() => useSyncSourcererUrlState(SourcererScopeName.default));
+    // Simulate onInitializeUrlParam call
+    const onInitializeUrlParam = mockUseInitializeUrlParam.mock.calls[0][1];
+    onInitializeUrlParam({ default: { id: 'test-id', selectedPatterns: ['a'] } });
+    expect(mockDispatch).not.toHaveBeenCalled();
+    expect(mockUpdateUrlParam).not.toHaveBeenCalled();
+  });
+
+  it('should dispatch setSelectedDataView if newDataViewPickerEnabled is false and initialState is provided', () => {
+    mockUseIsExperimentalFeatureEnabled.mockReturnValue(false);
+    renderHook(() => useSyncSourcererUrlState(SourcererScopeName.default));
+    const onInitializeUrlParam = mockUseInitializeUrlParam.mock.calls[0][1];
+    onInitializeUrlParam({ default: { id: 'test-id', selectedPatterns: ['a'] } });
+    expect(mockDispatch).toHaveBeenCalledWith({
+      payload: {
+        id: 'default',
+        selectedDataViewId: 'test-id',
+        selectedPatterns: ['a'],
+      },
+      type: 'x-pack/security_solution/local/sourcerer/SET_SELECTED_DATA_VIEW',
+    });
+  });
+
+  it('should update url param if newDataViewPickerEnabled is false and initialState is null', () => {
+    mockUseIsExperimentalFeatureEnabled.mockReturnValue(false);
+    mockUseSelector
+      .mockReturnValueOnce('test-id') // scopeDataViewId
+      .mockReturnValueOnce(['a']); // selectedPatterns
+    renderHook(() => useSyncSourcererUrlState(SourcererScopeName.default));
+    const onInitializeUrlParam = mockUseInitializeUrlParam.mock.calls[0][1];
+    onInitializeUrlParam(null);
+    expect(mockUpdateUrlParam).toHaveBeenCalledWith({
+      default: { id: 'test-id', selectedPatterns: ['a'] },
+    });
+  });
+});
+
+describe('useRestoreDataViewManagerStateFromURL', () => {
+  const mockUseIsExperimentalFeatureEnabled =
+    experimentalFeatures.useIsExperimentalFeatureEnabled as jest.Mock;
+  const mockUseInitializeUrlParam = globalQueryString.useInitializeUrlParam as jest.Mock;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUseInitializeUrlParam.mockImplementation((_key, cb) => cb);
+  });
+
+  it('should not call initDataViewSelection if newDataViewPickerEnabled is false', () => {
+    const initDataViewSelection = jest.fn();
+    mockUseIsExperimentalFeatureEnabled.mockReturnValue(false);
+    renderHook(() =>
+      useRestoreDataViewManagerStateFromURL(initDataViewSelection, SourcererScopeName.default)
+    );
+    const onInitializeUrlParam = mockUseInitializeUrlParam.mock.calls[0][1];
+    onInitializeUrlParam({ default: { id: 'test-id', selectedPatterns: ['a'] } });
+    expect(initDataViewSelection).not.toHaveBeenCalled();
+  });
+
+  it('should call initDataViewSelection for each scope if newDataViewPickerEnabled is true', () => {
```

Comment:
nit, but I would have this map over all of the available scopes, rather than just 2